### PR TITLE
Display User Polls.

### LINKED ISSUE #2: Display User Polls
Feature: Displays polls made by user so they can edit/delete them.

### PR DESCRIPTION
**Pull request by jduong09**
Goal: Display User polls on dashboard page.

### CODE REVIEW

File: root/backend/server/server.js [hunk 1]
```diff
@@ -55,8 +37,24 @@ app.post('/polls', async (req, res) => {
     }
 
     const poll = new Poll(pollData);
-    const result = await poll.save();
-    console.log(result);
+    await poll.save();
+  } catch(err) {
+    console.log(err);
```
Comment (ziggysauce) [inline root/backend/server/server.js:42]: at some point, we'll want to probably send an error status or some error messaging to the client for a better UX

File: root/backend/server/server.js [hunk 2]
```diff
@@ -55,8 +37,24 @@ app.post('/polls', async (req, res) => {
...
+        return {
+          title: poll.name,
+          location: poll.location
+        }
+      });
+    });
+    res.json({ response: polls });
+    res.end();
   } catch(err) {
     console.log(err);
```
Comment (ziggysauce) [inline root/backend/server/server.js:59]: same comment as above

File: root/client/src/components/user/dashboard.js [hunk 3]
```diff
@@ -1,18 +1,52 @@
...
+  const [polls, setPolls] = useState([]);
   const handleSubmit = async (e, formData) => {
     e.preventDefault();  
     await fetch('/polls', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(formData) })
   }
 
+  useEffect(() => {
+    const getPolls = async () => {
+      return await fetch('/polls/all').then(data => data.json()).then(result => result.response);
+    }
```
Comment (ziggysauce) [inline root/client/src/components/user/dashboard.js]: does the immediate function return not work here?

```suggestion
    const getPolls = async () => await fetch('/polls/all').then(data => data.json()).then(result => result.response);
```

File: root/client/src/components/user/dashboard.js [hunk 4]
```diff
@@ -1,18 +1,52 @@
...
+
+    try {
+      const polls = getPolls();
+      polls.then(response => setPolls(response));
+    } catch(error) {
+      console.log(error);
```
Comment (ziggysauce) [inline root/client/src/components/user/dashboard.js:19]: similar to my other comments, we'll eventually want to add some better client-facing error messaging for an improved UX

File: root/client/src/components/user/dashboard.js [hunk 5]
```diff
@@ -1,18 +1,52 @@
...
+      <header><h1>Hello this is Dashboard.</h1></header>
+      <main>
+        <div id="div-main">
+          <div id="div-main-header">
+            <h2>Polls</h2>
+            <button onClick={() => { setShowPollForm(!showPollForm)} }>Create</button>
+          </div>
+        </div>
+        {!showPollForm && <ul id="list-polls">{arrPolls}</ul>}
+        {showPollForm && <PollForm handleSubmit={handleSubmit} />}
```
Comment (ziggysauce) [inline root/client/src/components/user/dashboard.js]: might it be cleaner to just use a ternary here instead? (the suggestion below is untested)

```suggestion
        {showPollForm ? <PollForm handleSubmit={handleSubmit} /> : <ul id="list-polls">{arrPolls}</ul>}
```

**Comment by jduong09** [comment]
Going to push this branch to main in order to have a basic function of displaying polls.

**Pull request merged**